components/datetime/src/time_zone.rs

```diff
@@ -344,6 +638,17 @@ impl TimeZoneFormat {
         }
     }
 
+    pub fn format<W>(&self, w: &mut W, value: &impl TimeZoneInput) -> fmt::Result
```

Comment:
Discussion: With this function, if all of the formatters fail, you end up with an unchanged `W`, correct? Is this the behavior we want, or should we return an error if all the formatters fail?

Reply:
re: `With this function, if all of the formatters fail, you end up with an unchanged W, correct`: Yeah, that's right. 

re: `Is this the behavior we want, or should we return an error if all the formatters fail?`:
I think we should return an error instead. But the problem here is that we chain the methods together so the original error is lost. What would be a good way to raise error here? do you want to raise a general error or a specific error?

Reply:
I think we had considered having two functions, one of which returns `Result`/`Option` and one of which does not. I think we should try to avoid the behavior where we silently leave the output unchanged.

I'd suggest:

- `format_no_fallback` returns a `Result`/`Option`
- `format` calls `format_no_fallback`, printing the GMT format as the last resort

Reply:
This comment is missing in the change probably due to my 'bad' iterative rebase operations. 

But I want to mention that in the new change, I created a new method `format_no_fallback` which returns a `Result`. Also, I created a field `fallback_units` in `TimeZoneFormat` to keep track of order of fallbacks.

Meanwhile, I still want to keep the change, which is to take a `TimeZoneFormatOptions` while creating a `TimeZoneFormat`. The reason is that it aligns the ICU4X principle, which moves data preparation out of formatting so that formatting is fast enough. Although the existing fallback format options do not require data preparation, it allows us to add a fallback ,which needs data preparation, in the future.